File: old/D_Another_Problem_About_Dividing_Numbers.cpp
#include "bits/stdc++.h"
#ifndef ONLINE_JUDGE
#include "debug/debug.h"
#endif
using namespace std;
typedef long long ll;

vector<int> primeFactors(int n) {
    vector<int> primfac;
    for (int i = 2; i * i <= n; ++i) {
        while (n % i == 0) {
            primfac.push_back(i);
            n /= i;
        }
    }
    if (n > 1) {
        primfac.push_back(n);
    }
    return primfac;
}

void solve() {
    int a, b, k;
    cin >> a >> b >> k;

    vector<int> x = primeFactors(a);
    vector<int> y = primeFactors(b);

    map<int, int> ca, cb;
    for (int val : x) {
        ca[val]++;
    }
    for (int val : y) {
        cb[val]++;
    }

    set<int> seen;
    int need = 0, aa = 0, bb = 0;

    for (auto& [key, val] : ca) {
        if (seen.find(key) == seen.end()) {
            seen.insert(key);
            int w = val;
            int z = cb[key];
            if (w != z) {
                need++;
                int diff = abs(w - z);
                if (w > z) {
                    aa += diff;
                } else {
                    bb += diff;
                }
            }
        }
    }

    for (auto& [key, val] : cb) {
        if (seen.find(key) == seen.end()) {
            seen.insert(key);
            int w = val;
            int z = ca[key];
            if (w != z) {
                need++;
                int diff = abs(w - z);
                if (w > z) {
                    bb += diff;
                } else {
                    aa += diff;
                }
            }
        }
    }

    int tot = 0;
    if (aa) tot++;
    if (bb) tot++;

    if (k < tot || k > (int)(x.size() + y.size())) {
        cout << "NO\n";
        return;
    }
    if (tot == 0 && k == 1) {
        cout << "NO\n";
        return;
    }

    cout << "YES\n";
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t = 1;
    cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}

File: old/D_Another_Problem_About_Dividing_Numbers.py
import sys, math, random; from heapq import heapify, heappop, heappush;
from collections import defaultdict, Counter, deque; from bisect import bisect_left, bisect_right
def IL(): return list(map(int, sys.stdin.readline().strip().split()))
def S(): return sys.stdin.readline().strip()
def primeFactors(n):
    i = 2
    primfac = []
    while i * i <= n:
        while n % i == 0:
            primfac.append(i)
            n = n / i
        i = i + 1
    if n > 1:
        primfac.append(int(n))
    return primfac


def solve():
    a, b, k = IL()
    ran = random.randint(1000, 1000000000)
    x = primeFactors(a)
    y = primeFactors(b)
    ca = Counter()
    cb = Counter()
    for val in x:
        ca[val ^ ran] += 1
    for val in y:
        cb[val ^ ran] += 1

    seen = set()
    need = 0
    aa = bb = 0
    for key, val in ca.items():
        if key ^ ran not in seen:
            seen.add(key ^ ran)
            w = val
            z = cb[key]
            if w != z:
                need += 1
                diff = max(w, z) - min(w, z)
                if max(w, z) == w:
                    aa += diff
                else:
                    bb += diff
        
    for key, val in cb.items():
        if key ^ ran not in seen:
            seen.add(key ^ ran)
            w = val
            z = ca[key]
            if w != z:
                need += 1
                diff = max(w, z) - min(w, z)
                if max(w, z) == w:
                    bb += diff
                else:
                    aa += diff
    # if a == 1000000000:
    #     print(ca, cb)

    tot = 0
    if aa:
        tot += 1
    if bb:
        tot += 1
    # print(need, tot)

    if  k < tot or k > len(x) + len(y):
        print("NO")
        return
    # print(tot)
    if tot == 0 and k == 1:
        print("NO")
        return

    print("YES")
        
            
tt = 1
tt = IL()[0]
for _ in range(tt):
    solve()